.github/workflows/build.yaml

```diff
@@ -0,0 +1,107 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+name: Build
+on: push
+jobs:
+  build:
+    name: compile
+    runs-on: ubuntu-18.04
+    steps:
+        - uses: actions/checkout@master
+        - uses: ./.github/buildenv
+          with:
+             args: ./hadoop-ozone/dev-support/checks/build.sh
+  rat:
+    name: rat
+    runs-on: ubuntu-18.04
+    steps:
+        - uses: actions/checkout@master
+        - uses: ./.github/buildenv
+          with:
+             args: ./hadoop-ozone/dev-support/checks/rat.sh
+        - uses: actions/upload-artifact@master
+          if: always()
+          with:
+            name: rat
+            path: target/rat
+  author:
+    name: author
+    runs-on: ubuntu-18.04
+    steps:
+        - uses: actions/checkout@master
+        - uses: ./.github/buildenv
+          with:
+             args: ./hadoop-ozone/dev-support/checks/author.sh
+        - uses: actions/upload-artifact@master
+          if: always()
+          with:
+            name: author
+            path: target/author
+  unit:
+    name: unit
+    runs-on: ubuntu-18.04
+    steps:
+        - uses: actions/checkout@master
+        - uses: ./.github/buildenv
+          with:
+             args: ./hadoop-ozone/dev-support/checks/unit.sh
+        - uses: actions/upload-artifact@master
+          if: always()
+          with:
+            name: unit
+            path: target/unit
+  checkstyle:
+    name: checkstyle
+    runs-on: ubuntu-18.04
+    steps:
+        - uses: actions/checkout@master
+        - uses: ./.github/buildenv
+          with:
+             args: ./hadoop-ozone/dev-support/checks/checkstyle.sh
+        - uses: actions/upload-artifact@master
+          if: always()
+          with:
+            name: checkstyle
+            path: target/checkstyle
+  findbugs:
+    name: findbugs
+    runs-on: ubuntu-18.04
```

Comment:
Same here, should we serialize this behind the build ? that way if there is a compilation error we will see it and bail.

Reply:
I added dependencies between acceptance and others. I am not sure if we need other one here as findbugs is very fast it's not a big difference (in terms of resource usage) if findbugs are executed later.

But I am open to do it.

I would start the experiment and measure the resource usage and adjust the dependencies based on the data.